plugin-server/src/worker/ingestion/event-pipeline/1-populateTeamDataStep.ts

```diff
@@ -0,0 +1,31 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { PipelineEvent } from '../../../types'
+import { EventPipelineRunner, StepResult } from './runner'
+
+// TRICKY: team_id is optional on PipelineEvent but not for PluginEvent
+// Only do the type asserton to PluginEvent having verified team_id exists
+export async function populateTeamDataStep(runner: EventPipelineRunner, event: PipelineEvent): Promise<StepResult> {
```

Comment:
Probably worthwhile adding an integration test via plugin-server/tests/worker/ingestion/event-pipeline/event-pipeline-integration.test.ts

Reply:
And/or a functional test via https://github.com/PostHog/posthog/blob/master/plugin-server/functional_tests/e2e.test.ts . Let me know if you need any help with these @yakkomajuri , I think I'm the only one that's added these so keen to understand if there's any reason for that that I can help with.